libs/nucs/src/selector.rs

```diff
@@ -0,0 +1,97 @@
+use std::{collections::HashSet, str::FromStr, sync::LazyLock};
+
+static LABEL_ALPHABET: LazyLock<HashSet<char>> =
+    LazyLock::new(|| ('a'..='z').chain('A'..='Z').chain('0'..='9').chain(['_', '-']).collect());
```

Comment:
alphabet only allows `[a-zA-Z0-9_-]`, meaning selector like `.foo[0]` would fail, because `[` and `]` aren’t allowed